### PR TITLE
handle uncompressed chd files and unsupported codecs

### PR DESCRIPTION
**Pull request by inolen**

### CODE REVIEW

**Comment by inolen** [comment]
Wanted to bump this after seeing some other PRs come through.

File: src/chd.c [hunk 1]
```diff
@@ -789,6 +789,13 @@ chd_error cdfl_codec_decompress(void *codec, const uint8_t *src, uint32_t comple
     CODEC INTERFACES
 ***************************************************************************/
 
+#define CHD_MAKE_TAG(a,b,c,d)       (((a) << 24) | ((b) << 16) | ((c) << 8) | (d))
+
+/* general codecs with CD frontend */
+#define CHD_CODEC_NONE 0
+#define CHD_CODEC_CD_ZLIB CHD_MAKE_TAG('c','d','z','l')
+#define CHD_CODEC_CD_LZMA CHD_MAKE_TAG('c','d','l','z')
+#define CHD_CODEC_CD_FLAC CHD_MAKE_TAG('c','d','f','l')
```
Comment (baka0815) [inline src/chd.c]: As my changes got merged, those are in the header file. Only ``CHD_CODEC_NONE`` is missing and should be move there.
https://github.com/rtissera/libchdr/blob/6726f52a1d3ba04cebefade2bfcbb9dfc85eef00/src/chd.h#L207

File: src/chd.c [hunk 2]
```diff
@@ -1336,42 +1361,42 @@ chd_error chd_open_file(core_file *file, int mode, chd_file *parent, chd_file **
...
+			if (newchd->codecintf[decompnum]->init != NULL)
+			{
+				void* codec = NULL;
+				switch (newchd->header.compression[decompnum])
+				{
+					case CHD_CODEC_CD_ZLIB:
+						codec = &newchd->cdzl_codec_data;
+						break;
+
+					case CHD_CODEC_CD_LZMA:
```
Comment (baka0815) [inline src/chd.c]: As my changes got merged, you want to (re)add ``CHD_CODEC_ZLIB`` here.
See here for reference: https://github.com/rtissera/libchdr/blob/6726f52a1d3ba04cebefade2bfcbb9dfc85eef00/src/chd.c#L1359

File: src/chd.h [hunk 3]
```diff
@@ -237,8 +237,10 @@ extern "C" {
 #define CDROM_TRACK_METADATA_FORMAT	"TRACK:%d TYPE:%s SUBTYPE:%s FRAMES:%d"
 #define CDROM_TRACK_METADATA2_TAG	CHD_MAKE_TAG('C','H','T','2')
 #define CDROM_TRACK_METADATA2_FORMAT	"TRACK:%d TYPE:%s SUBTYPE:%s FRAMES:%d PREGAP:%d PGTYPE:%s PGSUB:%s POSTGAP:%d"
+
```
Comment (baka0815) [inline src/chd.h]: Intentional newline?

**Comment by baka0815** [comment]
I adopted your changes (see https://github.com/baka0815/reicast-emulator/commit/61afd4a8386987fd7d254acb32683f347d6226c8) and can confirm that with them uncompressed CHDv5 are working.

After the conflicts are resolved I don't see a problem merging this.

**Comment by inolen** [comment]
@rtissera I can rebase and push this again tonight if that'd make it easier.

**Comment by rtissera** [comment]
@inolen if you can rebase and push again that would be great.

**Comment by baka0815** [comment]
@inolen are you still working on this?

**Comment by inolen** [comment]
@rtissera sorry for the delay, this had completely slipped my mind.

Rebased and tested that everything still compiled - looks good to go.

**Comment by rtissera** [comment]
Merged, thanks !

**Pull request merged**